src/trace/tracer-wrapper.ts

```diff
@@ -99,4 +100,28 @@ export class TracerWrapper {
     this.tracer.inject(span, "text_map", dest);
     return dest;
   }
+
+  public setConsumeCheckpoint(contextJson: any, eventType: string, arn: string): void {
+    if (!arn) {
+      logDebug("DSM: No ARN provided, skipping setConsumeCheckpoint");
+      return;
+    }
+
+    if (!this.isDataStreamsEnabled) {
+      return;
+    }
+
+    try {
+      this.tracer.dataStreamsCheckpointer.setConsumeCheckpoint(eventType, arn, contextJson, false);
+    } catch (err) {
+      if (err instanceof Object || err instanceof Error) {
+        logDebug(`DSM: Failed to set consume checkpoint for ${eventType} ${arn}:`, err);
+      }
+    }
+  }
+
+  public get isDataStreamsEnabled(): boolean {
+    const validEnabledValues = new Set(["true", "1", "yes", "y", "on", "1"]);
+    return validEnabledValues.has(getEnvValue("DD_DATA_STREAMS_ENABLED", "false").toLowerCase());
+  }
```

Comment:
We should move this config check to the index.ts file, so it sits along with all the other configuration and means we only need to read the env var once.